dm/tests/safe_mode/run.sh

```diff
@@ -239,7 +298,7 @@ function run() {
 		# DM-worker1 is sharding lock owner and exits
 		if [ "$(check_port_return $WORKER1_PORT)" == "0" ]; then
 			echo "DM-worker1 is sharding lock owner and detects it offline"
-			export GO_FAILPOINTS='github.com/pingcap/tiflow/dm/syncer/SafeModeInitPhaseSeconds=return(0)'
```

Comment:
this failpoint is for closing safe mode in re-sync.

If dm needn't enable safe mode in re-sync, do we consider removing it?